javascript/frameworks/cap/lib/advanced_security/javascript/frameworks/cap/CQL.qll

```diff
@@ -3,20 +3,24 @@ import DataFlow
 import advanced_security.javascript.frameworks.cap.CDS
 
 /**
- * Objects from the SQL-like fluent API
- * this is the set of clauses that acts as the base of a statement
+ * Objects from the SQL-like fluent API that forms the basis of constructing
+ * a CQL clause.
  */
 class CqlQueryBase extends VarRef {
   CqlQueryBase() {
-    exists(string name |
-      this.getName() = name and
-      name in ["SELECT", "INSERT", "DELETE", "UPDATE", "UPSERT"] and
-      /* Made available as a global variable */
-      exists(GlobalVariable queryBase | this = queryBase.getAReference())
-      or
-      /* Imported from `cds.ql` */
-      exists(CdsFacade cds |
-        cds.getMember("ql").getMember(name).getAValueReachableFromSource().asExpr() = this
+    exists(VarRef varRef | this = varRef |
+      exists(string name |
+        varRef.getName() = name and
+        name in ["SELECT", "INSERT", "DELETE", "UPDATE", "UPSERT"] and
+        (
+          /* Made available as a global variable */
+          exists(GlobalVariable queryBase | this = queryBase.getAReference())
+          or
+          /* Imported from `cds.ql` */
+          exists(CdsFacade cds |
+            cds.getMember("ql").getMember(name).getAValueReachableFromSource().asExpr() = this
```

Comment:
Does this have to be a `VarRef`?

Reply:
Yes, `VarRef` was the most appropriate superclass of a function identifier (`` SELECT`col1, col2`.from`Table` ``) and a variable read (`` SELECT.from`Table`.columns`col1, col2` ``).